src/AutoRest.CSharp/Common/Generation/Writers/FormattableStringHelpers.cs

```diff
@@ -134,8 +127,33 @@ public static FormattableString GetConversionFormattable(this Parameter paramete
             return $"{parameter.Name:I}{conversionMethod}";
         }
 
+        private static FormattableString GetConversionFromFrameworkToRequestContent(this Parameter parameter, string? contentType)
+        {
+            if (TypeFactory.IsReadWriteDictionary(parameter.Type))
+            {
+                return $"{typeof(RequestContentHelper)}.{nameof(RequestContentHelper.FromDictionary)}({parameter.Name})";
+            }
+
+            if (TypeFactory.IsList(parameter.Type))
+            {
+                return $"{typeof(RequestContentHelper)}.{nameof(RequestContentHelper.FromEnumerable)}({parameter.Name})";
+            }
+
+            BodyMediaType? mediaType = contentType == null ? null : ToMediaType(contentType);
+            if (parameter.RequestLocation == RequestLocation.Body && mediaType == BodyMediaType.Binary)
+            {
+                return $"{parameter.Name:I}";
+            }
+            if (parameter.RequestLocation == RequestLocation.Body && mediaType == BodyMediaType.Text && parameter.Type.FrameworkType == typeof(string))
```

Comment:
what if the frameType is not string? e.g.  bool, int,  DatetimeOffset etc.  Are those valid case to support?

Reply:
No, we don't know how to support. We haven't discussed it.